source/adapters/hip/usm.cpp

```diff
@@ -72,7 +72,11 @@ USMFreeImpl([[maybe_unused]] ur_context_handle_t hContext, void *pMem) {
   try {
     hipPointerAttribute_t hipPointerAttributeType;
     UR_CHECK_ERROR(hipPointerGetAttributes(&hipPointerAttributeType, pMem));
-    unsigned int Type = hipPointerAttributeType.memoryType;
+#if HIP_VERSION_MAJOR >= 6
```

Comment:
Judging by the HIP headers and release docs, the `memoryType` field name has been deprecated since ROCm 5.6.0, so use of the field name `type` is encouraged since. I think it makes sense to guard in a #if HIP_VERSION >= 50600000. Nothing wrong with only using it from version 6.0 but I think the above suggestion will benefit us to avoid nesting ROCm > 5 with ROCm > 6 in some cases in the source code.

Reply:
Good point, agree. Amended the commit.

Reply:
nice